tests/main.tftest.hcl

```diff
@@ -0,0 +1,136 @@
+mock_provider "github" {
+  alias = "mock"
+}
+
+variables {
+  github_organization = "example-org"
+  teams = {
+    "developers" = {
+      name = "Developers"
+      members = [
+        {
+          username = "rachel"
+          role     = "maintainer"
+        },
+        {
+          username = "monica"
+          role     = "maintainer"
+        },
+        {
+          username = "ross"
+          role     = "member"
+        }
+      ]
+    },
+    "platform-engineers" = {
+      name    = "Platform Engineers"
+      privacy = "closed"
+      members = [
+        {
+          username = "chandler"
+          role     = "maintainer"
+        }
+      ]
+    }
+  }
+  repository_collaborators = {
+    "example-org/repo1" = [
+      {
+        username   = "monica"
+        permission = "admin"
+      },
+      {
+        username   = "joey"
+        permission = "push"
+      }
+    ],
+    "example-org/repo2" = [
+      {
+        username   = "phoebe"
+        permission = "maintain"
+      }
+    ]
+  }
+}
+
+run "validate_developers_team" {
+  command = plan
+
+  providers = {
+    github = github.mock
+  }
+
+  assert {
+    condition     = github_team.default["developers"].name == "Developers"
+    error_message = "Developers team should have correct name"
+  }
+
+  assert {
+    condition     = github_team.default["developers"].privacy == "secret"
+    error_message = "Developers team should have default privacy setting (secret)"
+  }
+}
+
+run "validate_platform_engineers_team" {
+  command = plan
+
+  providers = {
+    github = github.mock
+  }
+
+  assert {
+    condition     = github_team.default["platform-engineers"].name == "Platform Engineers"
+    error_message = "Platform Engineers team should have correct name"
+  }
+
+  assert {
+    condition     = github_team.default["platform-engineers"].privacy == "closed"
+    error_message = "Platform Engineers team should have overridden privacy setting (closed)"
+  }
+}
+
+run "validate_team_memberships" {
+  command = plan
+
+  providers = {
+    github = github.mock
+  }
```

Comment:
Aha this makes it so the GH provider is not initialized at all? So we're just testing the logic and not the underlying provider's logic?

Reply:
Yeah, we're using a completely mocked version of the GitHub provider, not the real one: no actual GitHub API calls are made, so no real resources are created/modified/deleted. I feel like this is safer, because we're not actually making any changes to GitHub org membership and team structure.
This still allows us to test the module's logic and transformations, and expected resource configurations. How do you feel about this approach?